### PR TITLE
change: separate logs() from attach()

### PR DESCRIPTION
**Pull request by ChuyangDeng**
*Issue #, if available:*
#1405 

*Description of changes:*
Disable logs for ``Estimator.attach()`` method and add a new ``logs()`` method.

*Testing done:*
Unit test.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/estimator.py [hunk 1]
```diff
@@ -630,9 +630,15 @@ def attach(cls, training_job_name, sagemaker_session=None, model_channel_name="m
             sagemaker_session=sagemaker_session, job_name=training_job_name
         )
         estimator._current_job_name = estimator.latest_training_job.name
-        estimator.latest_training_job.wait()
+        estimator.latest_training_job.wait(logs="None")
```
Comment (laurenyu) [inline src/sagemaker/estimator.py:633]: update the docstring at ll. 592-593 (where it talks about displaying logs)

File: src/sagemaker/estimator.py [hunk 2]
```diff
@@ -630,9 +630,15 @@ def attach(cls, training_job_name, sagemaker_session=None, model_channel_name="m
...
         return estimator
 
+    def logs(self):
+        """Display the logs for Estimator's training job. If the output is a tty or a Jupyter
```
Comment (metrizable) [inline src/sagemaker/estimator.py:637]: nit [pep-0257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings): single line, continue with detail after blank line

File: tests/unit/test_estimator.py [hunk 3]
```diff
@@ -681,6 +681,27 @@ def test_attach_framework(sagemaker_session):
...
+    sagemaker_session.sagemaker_client.describe_training_job = mock_describe_training_job
+    Estimator.attach(training_job_name="job", sagemaker_session=sagemaker_session)
+    sagemaker_session.logs_for_job.assert_not_called()
+
+
+def test_logs(sagemaker_session):
+    returned_job_description = RETURNED_JOB_DESCRIPTION.copy()
+    mock_describe_training_job = Mock(
+        name="describe_training_job", return_value=returned_job_description
+    )
```
Comment (metrizable) [inline tests/unit/test_estimator.py:698]: is this an opportunity for a fixture?

Comment (ChuyangDeng) [reply tests/unit/test_estimator.py:698]: Yea, there are in total 5 tests using this training job description

File: tests/unit/test_estimator.py [hunk 4]
```diff
@@ -191,6 +191,16 @@ def sagemaker_session():
     return sms
 
 
+@pytest.fixture()
```
Comment (laurenyu) [inline tests/unit/test_estimator.py:194]: is this per test or per module? (aka are there potential race conditions with the modification of this dictionary in the tests?)

Comment (ChuyangDeng) [reply tests/unit/test_estimator.py:194]: The pytest.fixture by default is run once per test function: https://docs.pytest.org/en/stable/fixture.html#scope-sharing-a-fixture-instance-across-tests-in-a-class-module-or-session

Each test is modifying the `RETURNED_JOB_DESCRIPTION.copy()` so should not cause race condition.

**Pull request merged**